### PR TITLE
Add pd facilitator program registration clientside components

### PR DESCRIPTION
**Pull request by Hamms**
Depends on #14300

### CODE REVIEW

File: apps/src/code-studio/pd/facilitator_program_registration/DateConfirm.jsx [hunk 1]
```diff
@@ -0,0 +1,178 @@
...
+        type: 'check'
+      });
+    }
+
+    return (
+      <FormGroup>
+        {followup}
+        <p>
+          Thank you for letting us know. You do not need to complete the rest of
+          this form. Please click to submit your form.
```
Comment (mehalshah) [inline apps/src/code-studio/pd/facilitator_program_registration/DateConfirm.jsx:27]: Please click what?

File: apps/src/code-studio/pd/facilitator_program_registration/FacilitatorProgramRegistration.jsx [hunk 2]
```diff
@@ -0,0 +1,199 @@
+import React from 'react';
```
Comment (mehalshah) [inline apps/src/code-studio/pd/facilitator_program_registration/FacilitatorProgramRegistration.jsx:1]: More generally speaking - we should consider making general form classes. This seems reusable

File: apps/src/code-studio/pd/facilitator_program_registration/PhotoRelease.jsx [hunk 3]
```diff
@@ -0,0 +1,49 @@
...
+          royalties whatsoever. I hereby release Code.org, its contractors,
+          its employees and any third parties involved in the creation or
+          publication of Code.org digital marketing, from liability for any
+          claims by me or any third party in connection with my participation.
+        </p>
+        {this.buildButtonsFromOptions({
+          name: 'photoRelease',
+          label: "I agree",
+          type: 'check',
+          required: false
```
Comment (mehalshah) [inline apps/src/code-studio/pd/facilitator_program_registration/PhotoRelease.jsx:40]: Isn't this true

File: apps/src/code-studio/pd/facilitator_program_registration/TravelInformation.jsx [hunk 4]
```diff
@@ -0,0 +1,136 @@
...
+          })}
+          {this.props.data.dietaryNeeds &&
+            (
+              this.props.data.dietaryNeeds.includes("Food allergy") ||
+              this.props.data.dietaryNeeds.includes("Other")
+            ) &&
+            this.buildFieldGroup({
+              name: "dietaryNeedsNotes",
+              componentClass: "textarea",
+              label: "Please explain (optional):",
```
Comment (mehalshah) [inline apps/src/code-studio/pd/facilitator_program_registration/TravelInformation.jsx]: Please provide details - "please explain" sounds too harsh

File: apps/src/code-studio/pd/facilitator_program_registration/PhotoRelease.jsx (hunk 3 above)
Comment (Hamms) [reply apps/src/code-studio/pd/facilitator_program_registration/PhotoRelease.jsx:40]: the liability waiver is required, the photo release is not (as per discussion in the bug bash)

File: apps/src/code-studio/pd/facilitator_program_registration/DateConfirm.jsx (hunk 1 above)
Comment (Hamms) [reply apps/src/code-studio/pd/facilitator_program_registration/DateConfirm.jsx:27]: This is a little weird, but it's the copy I was given, and it ends up rendering directly before the submit button: 

image

I think it works

File: apps/src/code-studio/pd/facilitator_program_registration/FacilitatorProgramRegistration.jsx (hunk 2 above)
Comment (Hamms) [reply apps/src/code-studio/pd/facilitator_program_registration/FacilitatorProgramRegistration.jsx:1]: Indeed! The intent of both this and the `ProgramRegistrationComponent` is that they should be able to be trivially converted into something like a `FormHandler` and `FormComponent` pair of classes, for generalizable form creation.

I just plan to do that generalization in a separate PR

File: apps/src/code-studio/pd/facilitator_program_registration/TravelInformation.jsx (hunk 4 above)
Comment (Hamms) [reply apps/src/code-studio/pd/facilitator_program_registration/TravelInformation.jsx]: done

**Pull request merged**